oap-server/server-library/library-client/src/main/java/org/apache/skywalking/oap/server/library/client/elasticsearch/ElasticSearchClient.java

```diff
@@ -119,6 +119,13 @@ public ElasticSearchClient(String clusterNodes,
     @Override
     public void connect() throws IOException, KeyStoreException, NoSuchAlgorithmException, KeyManagementException, CertificateException {
         List<HttpHost> hosts = parseClusterNodes(protocol, clusterNodes);
+        if (client != null) {
+            try {
+                client.close();
+            } catch (Throwable t) {
+                log.error("ElasticSearch client reconnection fails based on new config", t);
+            }
+        }
```

Comment:
Same thread visibility problem for `client`, if the monitor thread failed to detect the `client` is initialized below(in line 129), the `client` is not closed and left open forever

Reply:
Good point.